lib/spy/subroutine.rb

```diff
@@ -364,9 +365,14 @@ def get_spies(base_object, singleton_methods = true)
 
       # @private
       def get_spy_id(method)
-        return nil unless method.parameters[0].is_a?(Array)
-        id = method.parameters[0][1].to_s.sub!("__spy_args_", "")
-        id.to_i if id
+        return nil unless method.parameters[0].is_a?(Array) &&
```

Comment:
Sure, I added a slight modification to check the truthiness as well (skips nil and false).

```ruby
if method.parameters[0].is_a?(Array) && method.parameters[0][1]
  raw_id = method.parameters[0][1].to_s
  if raw_id.start_with?(SPY_ARGS_PREFIX)
    raw_id[SPY_ARGS_PREFIX.length..-1].to_i
  end
end
```